src/meta/src/stream/source_manager.rs

```diff
@@ -1136,15 +1141,28 @@ impl SourceManager {
     /// The command will first updates `SourceExecutor`'s splits, and finally calls `Self::apply_source_change`
     /// to update states in `SourceManager`.
     async fn tick(&self) -> MetaResult<()> {
-        let split_assignment = {
+        let mut split_assignment = {
             let core_guard = self.core.lock().await;
             core_guard.reassign_splits().await?
         };
 
+        let dup_assignment_flag = split_assignment
+            .iter_mut()
+            .map(|(_, assignment)| validate_assignment(assignment))
+            .reduce(|a, b| a && b)
```

Comment:
From my understanding, it should be `||` instead of `&&`. Please confirm?

Reply:
Oh yes, it is a bug. Let me fix.